cli/cmd/sync.go

```diff
@@ -195,6 +200,9 @@ func sync(cmd *cobra.Command, args []string) error {
 			managedplugin.WithTeamName(teamName),
 			managedplugin.WithLicenseFile(licenseFile),
 		}
+		if isThereADockerSourcePluginWithBackend {
```

Comment:
Shouldn't we do this only for destinations that are used as state?

Reply:
Also a follow PR can be to update the logic to consider if the OS is Windows per https://github.com/cloudquery/cloudquery/issues/18039

Reply:
I made the following changes:
- UseTCP is more strict; only enabling it when the destination is used as state
- I propagated the state that a source `shouldReplaceLocalhost` all the way to `ReplaceVariables`. Before, it was replacing if a source had a `RegistryDocker`, which is not ideal because it doesn't necessarily need to use state. I don't think it's a bug because if it doesn't use a backend it probably doesn't need any replacing, but it's more correct now.
- Added tests where possible.

Regarding the Windows edge case, I'd fix it separately (in another PR, as you say)